test/lib/mutation_source_test.cc

```diff
@@ -1419,6 +1419,28 @@ void test_slicing_with_overlapping_range_tombstones(tests::reader_concurrency_se
     }
 }
 
+void test_downgrade_to_v1_clear_buffer(tests::reader_concurrency_semaphore_wrapper& semaphore, populate_fn_ex populate) {
+    simple_schema s;
+    auto pkey = s.make_pkey();
+    const size_t row_count = 100; // Enough to trigger is_buffer_full.
```

Comment:
Could we calculate this based on buffer size so that changing the buffer size in the future wouldn't break this test?

Reply:
It would probably be nice, but this kind of tests requires to prepare all the mutations, in order to construct the reader. I could not use `run_mutation_reader_tests` and prepare a completely separate test, or I could increase the row_count by some amount (perhaps to the default value). The first solution also requires modifying the interface of flat_mutation_reader (and flat_mutation_reader_v2), since currently the variable controlling the maximum size of the buffer is marked private.

Reply:
Aren't you preparing the mutations yourself below?

Reply:
@denesb I am, however the reader is created by a function, based on the mutations I made:
```cpp
std::vector<mutation> mutations {{s.schema(), pkey}};
// ...
auto ms = populate(s.schema(), mutations, gc_clock::now());
assert_that(downgrade_to_v1(ms.make_reader_v2(s.schema(), semaphore.make_permit())))
// ...
```
I'm not sure what is the best solution here.

Reply:
You don't need to control the creation of the reader to pull this off. You can create your data such that it has more rows that what can fit into the default max buffer size.

Reply:
Done